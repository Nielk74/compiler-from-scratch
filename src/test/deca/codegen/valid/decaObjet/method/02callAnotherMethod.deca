// Description:
//    Test avec appels des methodes dans la classe
//
// Resultats:
//    02callAnotherMethod.deca
//
// Historique:
//    cree le 16/06/2022

class A {
    int x;

    void m() {
        d();
        println("this is method m");
    }

    void d(){
        f();
        println("this is method d");
    }

    void f(){
        println("this is method f");
    }

  int method1(int att1){
    if(att1 == 0){
      return method1(method1(1));
    }
    if(att1 == 1){
      return 2;
    }else{
      return 1;
    }
  }
}
{
    A a = new A();
    a.m();
    println(a.method1(0));

}