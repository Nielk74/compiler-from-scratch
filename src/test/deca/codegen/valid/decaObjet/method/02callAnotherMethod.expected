this is method f
this is method d
this is method m
1
